### ISSUE TITLE
CFE crashes when it receives an error message from AWS during failover

### ISSUE THREAD

**Issue by vkurenyshev-godaddy** [F5Networks/f5-cloud-failover-extension#80]
## Do you already have an issue opened with F5 support?
No

## Description
Failover fails when CFE gets an error response from AWS that request was not fulfilled.

Logs:
```
Fri, 09 Jul 2021 04:04:52 GMT - finest: [f5-cloud-failover] Function error, retrying: Request limit exceeded. Retries left: 50
Fri, 09 Jul 2021 04:04:52 GMT - severe: [f5-cloud-failover] error in _getFailoverDiscovery: RequestLimitExceeded: Request limit exceeded.
Fri, 09 Jul 2021 04:04:52 GMT - severe: [f5-cloud-failover] Request limit exceeded. RequestLimitExceeded: Request limit exceeded.
    at Request.extractError (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/services/ec2.js:50:35)
    at Request.callListeners (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/request.js:688:14)
    at Request.transition (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/request.js:690:12)
    at Request.callListeners (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
    at Request.emit (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/request.js:688:14)
    at Request.transition (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/request.js:690:12)
    at Request.callListeners (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
    at callNextListener (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/sequential_executor.js:96:12)
    at IncomingMessage.onEnd (/var/config/rest/iapps/f5-cloud-failover/node_modules/aws-sdk/lib/event_listeners.js:313:13)
    at emitNone (events.js:111:20)
    at IncomingMessage.emit (events.js:208:7)
Fri, 09 Jul 2021 04:04:52 GMT - finest: [f5-cloud-failover] Uploading data to: f5cloudfailover/f5cloudfailoverstate.json
Fri, 09 Jul 2021 04:04:52 GMT - finest: [f5-cloud-failover] Telemetry submitted successfully
Fri, 09 Jul 2021 04:04:57 GMT - finest: socket 189 closed
```

I guess I hit AWS API throttling limit as I see `Request limit exceeded. RequestLimitExceeded: Request limit exceeded.`.

There are several problems here:
1. Despite the message `Retries left: 50` CFE fails completely, and does not try to move IP addresses to another EC2 network interface.
2. F5 failover happens!!
  - we have a disconnect of AWS IP addresses configured on LB2 network interface; while
  - LB1 is active.

As a result -> FULL OUTAGE.

I also see the same behaviour when other problems happen.

In general, even CFE is trying to do something later, it executes F5 failover immediately without waiting for successful result from AWS.

## Environment information
For bugs, enter the following information:
- Cloud Failover Extension Version: 1.8.0
- BIG-IP version: 16.0.1
- Cloud provider: AWS
- CF template: Template v5.12.0

## Severity Level

Severity: 2

Severity level definitions:
 1. **Severity 1 (Critical)** : Defect is causing systems to be offline and/or nonfunctional. immediate attention is required.
 2. **Severity 2 (High)** : Defect is causing major obstruction of system operations.
 3. **Severity 3 (Medium)** : Defect is causing intermittent errors in system operations.
 4. **Severity 4 (Low)** : Defect is causing infrequent interuptions in system operations.
 5. **Severity 5 (Trival)** : Defect is not causing any interuptions to system operations, but none-the-less is a bug.

**Comment by shyawnkarim** [comment]
This issue is being tracked internally with ID AUTOSDK-556.

**Comment by shyawnkarim** [comment]
Closing.

This issue was addressed with [Release 1.10](https://github.com/F5Networks/f5-cloud-failover-extension/releases/tag/v1.10.0).
